src/WebJobs.Script/HostOptionsProvider.cs

```diff
@@ -0,0 +1,106 @@
+using Microsoft.Azure.WebJobs.Hosting;
+using Microsoft.Azure.WebJobs.Script.Configuration;
+using Microsoft.Azure.WebJobs.Script.Extensions;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Newtonsoft.Json.Serialization;
+
+namespace Microsoft.Azure.WebJobs.Script
+{
+    public class HostOptionsProvider : IHostOptionsProvider
+    {
+        private readonly JsonSerializerSettings _concurrencySettings = new JsonSerializerSettings
+        {
+            ContractResolver = new DefaultContractResolver
+            {
+                NamingStrategy = new CamelCaseNamingStrategy()
+            }
+        };
+
+        private readonly IEnumerable<IWebJobsExtensionOptionsConfiguration> _extensionOptionsConfigurations;
+        private readonly IServiceProvider _serviceProvider;
+        private readonly ILogger<HostOptionsProvider> _logger;
+        private readonly IOptionsMonitor<ConcurrencyOptions> _concurrencyOptions;
+
+        public HostOptionsProvider(IServiceProvider provider, IEnumerable<IWebJobsExtensionOptionsConfiguration> extensionOptionsConfigurations, IOptionsMonitor<ConcurrencyOptions> concurrencyOption, ILogger<HostOptionsProvider> logger)
+        {
+            _extensionOptionsConfigurations = extensionOptionsConfigurations;
+            _serviceProvider = provider;
+            _logger = logger;
+            _concurrencyOptions = concurrencyOption;
+        }
+
+        public string GetPayload()
+        {
+            var payload = new JObject();
+            var extensions = new JObject();
+            var extensionOptions = GetExtensionOptions();
+            foreach (var extension in extensionOptions)
+            {
+                extensions.Add(extension.Key, extension.Value);
+            }
+            if (extensionOptions.Count != 0)
+            {
+                payload.Add("extensions", extensions);
+            }
+
+            var concurrency = GetConcurrencyOption();
+            payload.Add("concurrency", concurrency);
+            return payload.ToString();
+        }
+
+        private Dictionary<string, JObject> GetExtensionOptions()
+        {
+            var configuration = _serviceProvider.GetService<IConfiguration>();
+            // for each of the extension options types identified, create a bound instance
+            // and format to JObject
+            Dictionary<string, JObject> result = new Dictionary<string, JObject>();
+            foreach (IWebJobsExtensionOptionsConfiguration extensionOptionsConfigration in _extensionOptionsConfigurations)
+            {
+                // create the IOptions<T> type
+                Type optionsWrapperType = typeof(IOptions<>).MakeGenericType(extensionOptionsConfigration.OptionType);
+                object optionsWrapper = _serviceProvider.GetService(optionsWrapperType);
```

Comment:
Consider some error handling here. E.g. could the GetService call returns null?

Reply:
Fixed.